geomstats/learning/mdm.py

```diff
@@ -94,6 +95,31 @@ def predict(self, X):
             y.append(self.classes_[index])
         return gs.array(y)
 
+    def predict_proba(self, X):
+        """Compute probabilities to belong to classes according to
+        riemannian_metric.
+
+        Parameters
+        ----------
+        X : array-like, shape=[n_samples, dim]
+                              if point_type='vector'
+                              shape=[n_samples, n, n]
+                              if point_type='matrix'
+            Test data, where n_samples is the number of samples
+            and n_features is the number of features.
+
+        Returns
+        -------
+        probas : array-like, shape=[n_samples, n_classes]
+            Probability of the sample for each class in the model.
+        """
+        n_samples = X.shape[0]
+        probas = []
+        for i in range(n_samples):
+            dists = self.riemannian_metric.dist(X[i], self.mean_estimates_)
```

Comment:
From an efficiency perspective, maybe it would make more sense to use `riemannian_metric.squared_dist` here. Most often, the Riemannian square distance is computed using the inner-product of the Riemannian Log with itself, and then the Riemannian distance is computed by taking the square of this:
https://github.com/geomstats/geomstats/blob/efe8f0215df802f4dd2816fcac42bd81b7a900d7/geomstats/geometry/riemannian_metric.py#L200


If we directly use the .squared_dist here, we spare one square-root and one square in the default case, which could be interesting.

Reply:
Yes, you are right! I will correct this point.